### PR TITLE
Android

### PR DESCRIPTION
**Pull request by todi1856**

### CODE REVIEW

**Comment by todi1856** [comment]
Git is so confusing for me... Don't review yet... for some reason it was showing me 1000 files changed, but apparently it was because there were conflicts

**Comment by todi1856** [comment]
This PR should be good to go again, pls note, due some weirdness with variants, gamepad tests will fail.

@Rene-Damm can you review again pls, thank you

File: Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidSensorTests.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            typeof(AndroidRelativeHumidity),
+            typeof(AndroidAmbientTemperature),
+            typeof(AndroidMagneticFieldUncalibrated),
+            typeof(AndroidGameRotationVector),
+            typeof(AndroidGyroscopeUncalibrated),
+            typeof(AndroidSignificantMotion),
+            typeof(AndroidStepDetector),
+            typeof(AndroidStepCounter),
+            typeof(AndroidGeomagneticRotationVector),
+            typeof(AndroidHeartRate)
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidSensorTests.cs:35]: Use `TestCase` attribute for this scenario.

**Review by Rene-Damm — APPROVED** [review]

File: Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidTests.cs [hunk 2]
```diff
@@ -6,19 +6,51 @@
...
+            capabilities = new AndroidDeviceCapabilities
+            {
+                inputSources = AndroidInputSource.Joystick
+            }.ToJson()
+        };
+    }
+
+    [Test]
+    [Category("Devices")]
+    public void Devices_CorrectControllerTypeIsCreated()
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidTests.cs:38]: Avoid "correct". Specify behavior exactly. "Correct" is ambiguous.

File: Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidSensorTests.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            capabilities = new AndroidSensorCapabilities()
+            {
+                sensorType = type
+            }.ToJson()
+        };
+    }
+
+    [Test]
+    [Category("Devices")]
+    public void Devices_AccelerometerWorks()
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidSensorTests.cs:62]: Avoid "works"-style tests. Have tests specify more or less exactly what they're after.

Comment (Rene-Damm) [reply Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/AndroidSensorTests.cs:62]: (my own "SupportsX" kinda straddle that border...)

**Pull request merged**